feat: initialise monorepo (#290)
tuist/Noora: README.md

# Noora Design System ⭐️

Noora is Tuist's comprehensive design system that provides consistent UI components and patterns across the web and the CLI.

<!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->

[![All Contributors](https://img.shields.io/badge/all_contributors-3-orange.svg?style=flat-square)](#contributors-)

<!-- ALL-CONTRIBUTORS-BADGE:END -->

> [!NOTE]
> The project is currently in an early stage of development. Our plan is to create a basic set of components and then iterate on them based on the feedback we receive from the community.

## Domains

Noora is implemented for CLIs and the web to achieve cohesive design across different environments. Noora for CLI is implemented as a Swift package whereas we chose Elixir and Phoenix for the web.

> ### 📱 CLI (`packages/cli/`)

Reviewer: Actually, I would consider the link here to be to the documentation page. And then would mention the documentation page again after the header, such as by having a [highlighted section for it](https://github.com/tuist/tuist?tab=readme-ov-file#-documentation) or at least make it prominent with some emojis.

# Noora Design System

> Noora is Tuist's comprehensive design system that provides consistent UI components and patterns across all platforms and environments.

Reviewer: ```suggestion
Noora is Tuist's comprehensive design system that provides consistent UI components and patterns across the web and the CLI.
```

<!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->

[![All Contributors](https://img.shields.io/badge/all_contributors-3-orange.svg?style=flat-square)](#contributors-)

<!-- ALL-CONTRIBUTORS-BADGE:END -->

## Domains

> This monorepo contains multiple packages that implement Noora's design principles:

Reviewer: Noora is implemented for CLIs and the web to achieve cohesive design across different environments. Noora for CLI is implemented as a Swift package whereas we chose Elixir and Phoenix for the web.

# Noora Design System

Noora is Tuist's comprehensive design system that provides consistent UI components and patterns across all platforms and environments.

## Packages

> This monorepo contains multiple packages that implement Noora's design principles:

Reviewer: Instead of talking about `Packages`, I would talk about `Domains` – we have a web domain and a CLI domain.

> ### 📱 CLI Package (`packages/cli/`)

Reviewer: ```suggestion
### 📱 CLI (`cli/`)
```

A Swift package providing terminal UI components for building beautiful command-line interfaces.

### 🌐 Web Package (`packages/web/`) _(Coming Soon)_

A package to build interactive user interfaces for the web using Phoenix LiveView.

## Development

### Prerequisites

- [mise](https://mise.jdx.dev/) for tool management
- Swift 6.0+ (for CLI package)

> - Node.js 22+ (for web package)

Reviewer 2: Since this information is already in the `mise.toml`. Should we just point people to running `mise install` or checking the dependencies in that file?
